hawc/apps/common/serializers.py

```diff
@@ -283,3 +286,164 @@ def run_validators(self, value):
     def create(self, validated_data, *args, **kwargs):
         instance, created = self.Meta.model.objects.get_or_create(**validated_data)
         return instance
+
+
+class BulkSerializer(serializers.ListSerializer):
+    """
+    Bulk create/update serializer for many = True
+    "id" must be given write access on serializer for bulk update
+    (eg.  id = IntegerField(required=False))
+    """
+
+    def to_internal_value(self, data):
+        """
+        This is the inherited method from ListSerializer, with a few lines added to work around an error.
+        `exclude_current_instance` on `UniqueValidator` examines the `instance` set on the serializer to exclude it from the uniqueness constraint:
+        https://github.com/encode/django-rest-framework/blob/3875d3284e73ed4d8e36c07d9b70c1b22c9d5998/rest_framework/validators.py#L138
+        However, `instance` in this case is the queryset passed in to the constructor, so it throws an error when it attempts to retrieve the `pk` attribute of the queryset.
```

Comment:
minor nit; hard to read b/c the line is so long; no hard rule but I try to keep <100